lib/wordcounter.rb

```diff
@@ -0,0 +1,93 @@
+require 'delegate'
+
+class WordHistogram < SimpleDelegator
+  def initialize(*args)
+    super(Hash.new(0, *args))
+  end
+
+  def to_pretty_s
+    word_freq_pairs = self.sort_histogram
+
+    word_lengths = word_freq_pairs.map { |word, _| word }.map &:length
+    max_length = word_lengths.max
+
+    self.to_table_s(word_freq_pairs, middle_column_width: max_length)
+  end
+
+  protected
+  def sort_histogram
+    self.sort do |pair1, pair2|
+      word1, freq1 = *pair1
+      word2, freq2 = *pair2
+
+      next freq2 <=> freq1 unless freq2 == freq1
+      word1 <=> word2
```

Comment:
The issue is I went a step further and specifically wanted to sort by `freq` then by `word`. That way they are ranked by frequency and those that have the same frequency are sorted lexicographically. It was a minor stylistic thing. I tried `sort_by { |word, freq| [freq, word] } but I also need to read the array in reverse (or `reverse` it in my implementation) which then puts the words in reverse lexicographic ordering. 

With this implementation I can inversely compare the frequencies and regularly compare the words, giving the desired effect without a call to `reverse` or reading backwards. 

Evidently, `sort` will not explode both pairs in the parameters to the block, so I had to explode them myself.